R/snapshots.R

```diff
@@ -0,0 +1,116 @@
+#' Run Simulations From Snapshot Files
+#'
+#' @param ... character strings, path to snapshot files or a directory containing snapshot files 
+#' @param output character string, path to the output directory where to write simulation results
+#' @param csv logical, whether to export the results as csv (default = TRUE)
+#' @param pkml logical, whether to export the results as pkml (default = FALSE)
+#' @param xml logical, whether to export the results as xml (default = FALSE)
+#'
+#' @return NULL
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' runSimulationsFromSnapshot("path/to/my_snapshot.json", csv = TRUE, pkml = TRUE)
+#' }
+runSimulationsFromSnapshot <- function(..., output = ".", csv = TRUE, pkml = FALSE, xml = FALSE){
+  
+  initPKSim()
+
+  temp_dir <- .gatherFiles(c(...))
+  
+  JsonRunOptions <- rSharp::newObjectFromName("PKSim.CLI.Core.RunOptions.JsonRunOptions")
+  JsonRunOptions$set("InputFolder", temp_dir)
+  JsonRunOptions$set("OutputFolder", normalizePath(output))
+
+  if (isTRUE(csv)){ csv <- 2L } else { csv <- 0L }
+  if (isTRUE(xml)){ xml <- 4L } else { xml <- 0L }
+  if (isTRUE(pkml)){ pkml <- 8L } else { pkml <- 0L }
+  
+  ExportMode <- csv + xml + pkml
+  
+  JsonRunOptions$set("ExportMode", ExportMode)
+  #0: nothing
+  #1: json
```

Comment:
I don't see where **json** is added to `ExportMode`

Reply:
As we **input** `json` files for this function, it did not make any sense to allow user to output `.json` files

Reply:
These are not the same json
Input file are snapshot
Output files are structured simulation results

Reply:
Ok, thanks for clarification